pkg/store/bucket.go

```diff
@@ -874,14 +874,21 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 	req.MaxTime = s.limitMaxTime(req.MaxTime)
 
 	var (
-		ctx     = srv.Context()
-		stats   = &queryStats{}
-		res     []storepb.SeriesSet
-		mtx     sync.Mutex
-		g, gctx = errgroup.WithContext(ctx)
-		hints   = &hintspb.SeriesResponseHints{}
+		ctx      = srv.Context()
+		stats    = &queryStats{}
+		res      []storepb.SeriesSet
+		mtx      sync.Mutex
+		g, gctx  = errgroup.WithContext(ctx)
+		reqHints = &hintspb.SeriesRequestHints{}
+		resHints = &hintspb.SeriesResponseHints{}
 	)
 
+	if s.enableSeriesHints && req.Hints != nil {
+		if err := types.UnmarshalAny(req.Hints, reqHints); err != nil {
+			return status.Error(codes.InvalidArgument, errors.Wrap(err, "unmarshal series request hints").Error())
+		}
+	}
```

Comment:
Shouldn't unknown hints be simply ignored?

Reply:
I don't think so. If a client sends a request, the request should be valid. If it's invalid, it's a client issue and it should be fixed.

Reply:
If the request hints contains the wrong protobuf message, the unmarshalling will fail and it's expected (I've added a test now). If it contains the right message but with some extra unknown fields, that's fine.